src/main/java/com/synopsys/integration/alert/web/config/JobConfigActions.java

```diff
@@ -208,63 +209,59 @@ public String validateJob(final JobFieldModel jobFieldModel) throws AlertFieldEx
-                // TODO Create a mechanism to create the field accessor with a combination of fields in the database and fields that are not.
-                Optional<ConfigurationFieldModel> topicField = convertFieldToConfigurationField(channelFieldModel, TestAction.KEY_CUSTOM_TOPIC);
-                Optional<ConfigurationFieldModel> messageField = convertFieldToConfigurationField(channelFieldModel, TestAction.KEY_CUSTOM_MESSAGE);
-                topicField.ifPresent(model -> fields.put(TestAction.KEY_CUSTOM_TOPIC, model));
-                messageField.ifPresent(model -> fields.put(TestAction.KEY_CUSTOM_MESSAGE, model));
                 final TestAction testAction = testActionOptional.get();
-                final FieldAccessor fieldAccessor = new FieldAccessor(fields);
-                final String jobId = channelFieldModel.getId();
+                final String jobId = jobFieldModel.getJobId();
 
                 testProviderConfig(fieldAccessor, jobId, destination);
                 MessageResult testResult = testAction.testConfig(jobId, destination, fieldAccessor);
                 return testResult.getStatusMessage();
             } else {
-                final String descriptorName = channelFieldModel.getDescriptorName();
-                logger.error("Test action did not exist: {}", descriptorName);
-                throw new AlertMethodNotAllowedException("Test functionality not implemented for " + descriptorName);
+                logger.error("Test action did not exist: {}", channelDescriptorName);
+                throw new AlertMethodNotAllowedException("Test functionality not implemented for " + channelDescriptorName);
             }
         }
         return "No field model of type channel was was sent to test.";
     }
 
-    private FieldModel getChannelFieldModelAndPopulateOtherJobModels(JobFieldModel jobFieldModel, Collection<FieldModel> otherJobModels) throws AlertException {
-        FieldModel channelFieldModel = null;
-        for (final FieldModel fieldModel : jobFieldModel.getFieldModels()) {
-            final Optional<Descriptor> descriptor = descriptorProcessor.retrieveDescriptor(fieldModel.getDescriptorName());
-            final FieldModel updatedFieldModel = fieldModelProcessor.createCustomMessageFieldModel(fieldModel);
-            if (descriptor.filter(foundDescriptor -> DescriptorType.CHANNEL.equals(foundDescriptor.getType())).isPresent()) {
-                channelFieldModel = updatedFieldModel;
-            } else {
-                otherJobModels.add(updatedFieldModel);
-            }
+    private Optional<FieldModel> getCustomMessageFieldModelAndRemoveIt(JobFieldModel jobFieldModel) {
+        Optional<FieldModel> customFieldModel = jobFieldModel.getFieldModels()
+                                                    .stream()
+                                                    .filter(fieldModel -> fieldModel.getDescriptorName().endsWith(CUSTOM_MESSAGE))
```

Comment:
Why are we using the descriptor name here? Didn't we just explicitly remove all assumptions about the contents of descriptor names with the ChannelKey stuff?

Reply:
It looks like all of this is designed to send custom messages to channels. Each job can only have one channel, so why don't we just look for the channel section of the job model instead appending a string and then stripping it out again later?